tests/integration/helpers.py

```diff
@@ -231,3 +250,41 @@ def is_relation_broken(ops_test: OpsTest, endpoint_one: str, endpoint_two: str)
         if endpoint_one not in endpoints and endpoint_two not in endpoints:
             return True
     return False
+
+
+def is_connection_possible(credentials: Dict) -> bool:
+    """Test a connection to a MySQL server.
+
+    Args:
+        credentials: A dictionary with the credentials to test
+    """
+    config = {
+        "user": credentials["username"],
+        "password": credentials["password"],
+        "host": credentials["host"],
+        "raise_on_warnings": False,
+    }
+
+    try:
+        with MysqlConnector(config) as cursor:
+            cursor.execute("SELECT 1")
+            return cursor.fetchone()[0] == 1
+    except Exception as e:
+        print(e)
+        return False
+
+
+def instance_ip(model: str, instance: str) -> str:
+    """Translate juju instance name to IP.
+
+    Args:
+        model: The name of the model
+        instance: The name of the instance
+    Returns:
+        The (str) IP address of the instance
+    """
+    output = subprocess.check_output(f"juju machines --model {model}".split())
```

Comment:
minor nit: it may be possible to avoid using `subprocess.check_output` (and instead do `ops_test.juju("machines", "--model", f"{model}"`)

Reply:
That's was my first approach, but it was failing due to a bug in python libjuju, so I've just change to this